packages/flutter/lib/src/painting/linear_border.dart

```diff
@@ -0,0 +1,376 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+
+import 'basic_types.dart';
+import 'borders.dart';
+import 'edge_insets.dart';
+
+/// Defines the relative size and alignment of one <LinearBorder> edge.
+///
+/// A [LinearBorder] defines a box outline as zero to four edges, each
+/// of which is rendered as a single line. The width and color of the
+/// lines is defined by [LinearBorder.side].
+///
+/// Each line's length is defined by [size], a value between 0 and 1
+/// (the default) which defines the length as a percentage of the
+/// length of a box edge.
+///
+/// When [size] is less than 1, the line is aligned within the
+/// available space according to [alignment], a value between -1 and
+/// 1.  The default is 0, which means centered, -1 means align on the
+/// "start" side, and 1 means align on the "end" side. The meaning of
+/// start and end depend on the current [TextDirection], see
+/// [Directionality].
+@immutable
+class LinearBorderEdge {
+  /// Defines one side of a [LinearBorder].
+  ///
+  /// The values of [size] and [alignment] must be between
+  /// 0 and 1, and -1 and 1 respectively.
+  const LinearBorderEdge({
+    this.size = 1,
+    this.alignment = 0,
+  }) : assert(size >= 0 && size <= 1);
+
+  /// A value between 0 and 1 that defines the length of the edge as a
```

Comment:
nit: This paragraph seems like it could be re-flowed to be two lines instead of three.